dif-data-classes/src/commonMain/kotlin/at/asitplus/dif/rqes/CollectionEntries/Document.kt

```diff
@@ -0,0 +1,87 @@
+package at.asitplus.dif.rqes.CollectionEntries
+
+import at.asitplus.dif.rqes.Enums.ConformanceLevelEnum
+import at.asitplus.dif.rqes.Enums.SignatureFormat
+import at.asitplus.dif.rqes.Enums.SignedEnvelopeProperty
+import at.asitplus.signum.indispensable.asn1.ObjectIdentifier
+import kotlinx.serialization.EncodeDefault
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+/**
+ * CSC: Class used as part of [SignatureRequestParameters]
+ */
+@Serializable
+data class Document(
+    /**
+     * base64-encoded document content to be signed, testcases weird so for now string
+     */
+    @SerialName("document")
+//    @Serializable(ByteArrayBase64Serializer::class)
+    val document: String,
+
+    /**
+     * Requested Signature Format
+     */
+    @SerialName("signature_format")
+    val signatureFormat: SignatureFormat,
+
+    /**
+     * Requested conformance level. If omitted its value is "Ades-B-B"
```

Comment:
Maybe set this as the default value?

Reply:
If we set a default value we have a problem with the test cases as either 
A) we encode defaults, in which case the test case throws if no signature format was given
B) we do not encode defaults, in which case the test case throws if the value given matches the default value

I know that changing code based on test cases is not a great idea but I have not found a good solution to this problem yet